### ISSUE TITLE
does rawaccel support snapping angles?

### ISSUE THREAD

**Issue by owlw** [a1xd/rawaccel#175]
title

**Comment by Fuwn** [comment]
If you take a look at [this](https://github.com/a1xd/rawaccel/commit/8dac6b3ff1d3fa434c4cd1db752ba34681cae8b4) pull request, it looks like angle snapping was implemented at some point but never exposed in the public-facing GUI.

The implementation suggests that if you set the `degrees_snap` value within the `settings.json` file (adjacent to your rawaccel executable) to a non-zero, floating-point value, then the angle snapping will kick in and function similar to other software-based angle snapping implementations.

**Comment by HatsukuLive** [comment]
> If you take a look at [this](https://github.com/a1xd/rawaccel/commit/8dac6b3ff1d3fa434c4cd1db752ba34681cae8b4) pull request, it looks like angle snapping was implemented at some point but never exposed in the public-facing GUI.
> 
> The implementation suggests that if you set the `degrees_snap` value within the `settings.json` file (adjacent to your rawaccel executable) to a non-zero, floating-point value, then the angle snapping will kick in and function similar to other software-based angle snapping implementations that feature granular, angle predefinition.

True, it is there but for some reason, there's no way of making sure the settings.json are being effectively applied, I think there is a specific order of doing things so it works (haven't figured it out yet) because I need to do it multiple times in different ways for it to actually work.

**Comment by Fuwn** [comment]
> ... there's no way of making sure the settings.json are being effectively applied ...

Where'd you get that sentiment from?

The application of `settings.json` ultimately comes down to [`writer.exe`](https://github.com/a1xd/rawaccel/blob/master/writer/Program.cs) converting your `settings.json` to a [`DriverConfig`](https://github.com/a1xd/rawaccel/blob/master/wrapper/wrapper.cpp#LL474C18-L474C30) and writing it to the driver (using [`Activate`](https://github.com/a1xd/rawaccel/blob/master/wrapper/wrapper.cpp#LL490C12-L490C12))

For example, navigate to your Raw Accel install directory, open your `settings.json` in any text editor, and change the "Sensitivity multiplier" field (near the bottom) to double what it is now; and save it. Nothing should have happened. Now, open a command prompt within the directory and execute `writer.exe .\settings.json`. You should have noticed that your cursor speed is now double what it was before you executed the command.

One thing to note, though, is that upon opening the Raw Accel GUI (`rawaccel.exe`), your `settings.json` is automatically applied, regardless of you having had not clicked the "Apply" button manually. Additionally, while the GUI is open, anytime you click the apply button, the settings within the Raw Accel GUI will be forcefully saved to the `settings.json` file and written to the driver.

> ... I think there is a specific order of doing things ...

The moral of the story is that the GUI takes precedence over the configuration in almost all cases. If you would like to manually modify your `settings.json` file, do so, but activate it with `writer.exe` rather than the GUI (`rawaccel.exe`).

**Comment by HatsukuLive** [comment]
Oh you're right! Thank you for the quick response! 😄
